src/terminal/kitty.rs

```diff
@@ -32,11 +32,24 @@ use std::process::{Command, Stdio};
 use std::str;
 use url::Url;
 
-/// Whether we run in Kitty or not.
-pub fn is_kitty() -> bool {
+/// Whether we run in Kitty or not. It also returns the version.
+pub fn is_kitty() -> Option<(u8, u8, u8)> {
     std::env::var("TERM")
-        .map(|value| value == "xterm-kitty")
-        .unwrap_or(false)
+        .ok()
+        .filter(|value| value == "xterm-kitty")?;
+
+    let output = Command::new("kitty").arg("--version").output().ok()?;
```

Comment:
This will fail if `kitty` is not in `$PATH` or doesn't exist at all (e.g. inside a chroot or systemd container).   And worse, it won't fail gracefully in this case and detect kitty without hyperlinks; instead it'll make mdcat believe that it's not running in kitty at all, even if `$TERM` says it's kitty.

Can we somehow get the version in a more reliable way?  Perhaps by means of some escape sequence?

Reply:
If `kitty` is not in `$PATH` image display would also fail. You're right about `allow_hyperlinks`, but looking at the issue and at the commits in `kitty` I haven't seen anything implemented for feature detection.

Reply:
I wasn't aware that image display uses `kitty`.  In this case I agree that it's a good idea to fail completely if `kitty` isn't found!  But please document this in the docstring of the method, so that I don't forget and think it's a bug in six months :wink: